core/src/main/scala/core/database/LiftAsync.scala

```diff
@@ -0,0 +1,22 @@
+package com.zengularity.querymonad.core.database
+
+// import scala.concurrent.{ExecutionContext, Future}
+import scala.language.higherKinds
+
+/**
+ * Heavily inspired from work done by @cchantep in Acolyte (see acolyte.reactivemongo.ComposeWithCompletion)
+ */
+trait LiftAsync[F[_], M[_], A] {
```

Comment:
Semantic is always async ?

Reply:
yes (but no in term of `cats-effect`'s `Async` typeclass), since the essence of this typeclass is to wrap the computation in an async context (`Future` or `IO`). Maybe it can be confusing, so feel free to suggest any name which could express it better.

Reply:
`Task` ? `Effect` ?